src/main/scala/rest/UserRoutes.scala

```diff
@@ -0,0 +1,33 @@
+package rest
+
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import service.UserService
+import validation.Validator
+import validation.Validator.nonEmpty
+
+class UserRoutes(userService: UserService) extends JsonSupport {
+
+  private val userValidator: Validator[NewUser] = user => {
+    val validUsername = nonEmpty.validate(user.username)
+    if (validUsername.isLeft) Left(s"username ${validUsername.left.get}")
```

Comment:
the requirement is:
* username: String (should be non-empty and contain only alphanumeric characters, validate it)